pkg/util/inplaceupdate/inplace_update.go

```diff
@@ -227,10 +230,24 @@ func (c *realControl) finishGracePeriod(pod *v1.Pod, opts *UpdateOptions) (time.
 				return nil
 			}
 
-			if clone, err = opts.PatchSpecToPod(clone, &spec, &updateState); err != nil {
+			var expectedResources map[string]*v1.ResourceRequirements
+			clone, expectedResources, err = opts.PatchSpecToPod(clone, &spec, &updateState)
+			if err != nil {
 				return err
 			}
 			appspub.RemoveInPlaceUpdateGrace(clone)
+			if utilfeature.DefaultFeatureGate.Enabled(features.InPlaceUpdateResourceByResize) && len(expectedResources) != 0 {
```

Comment:
Hi, @chengjoey.
I've had a change of mind. I now believe that using client.GetCurrentServerVersion() to switch between the two branches is a more effective approach than introducing the InPlaceUpdateResourceByResize feature gate.

Reply:
Yes, using client.GetCurrentServerVersion() might be better, as this should not be an option for the user.